frontend/packages/kubevirt-plugin/integration-tests-cypress/views/dialog.ts

```diff
@@ -26,24 +27,37 @@ export const addNIC = (nic: Network) => {
 };
 
 export const addDisk = (disk: Disk) => {
-  cy.get(diskDialog.addDisk).click();
+  cy.get(disksTab.addDiskBtn).click();
   if (disk.source) {
     cy.get(diskDialog.source).click();
     cy.get('.pf-c-select__menu-item-main')
       .contains(disk.source.getDescription())
       .click();
-    cy.get('body').then(($body) => {
-      if ($body.find(diskDialog.diskURL).length) {
-        cy.get(diskDialog.diskURL)
-          .clear()
-          .type(disk.source.getSource());
-      }
-      if ($body.find(diskDialog.diskContainer).length) {
-        cy.get(diskDialog.diskContainer)
-          .clear()
-          .type(disk.source.getSource());
-      }
-    });
+    const sourceUrl = disk.source.getSource();
+    switch (disk.source) {
+      case ProvisionSource.URL:
+        if (sourceUrl) {
+          cy.get(diskDialog.diskURL).type(sourceUrl);
+        } else {
+          throw new Error('No `disk.source value` provided!!!');
+        }
+        break;
+      case ProvisionSource.REGISTRY:
+        if (sourceUrl) {
+          cy.get(diskDialog.diskContainer).type(sourceUrl);
+        } else {
+          throw new Error('No `disk.source value` provided!!!');
+        }
+        break;
+      case ProvisionSource.EXISTING:
+      case ProvisionSource.CLONE_PVC:
+        cy.get(diskDialog.diskPVC).select(disk.pvcName);
+        break;
+      case ProvisionSource.EPHEMERAL:
```

Comment:
Need to type `sourceURL` for EPHEMERAL as well.

Reply:
We do not have this kind of ProvisionSource in disks, and I can't find it's usage, so I think it was just redundant alias for REGISTRY. Nevertheless I put it same as REGISTRY, since it has same `.getSource()` output.